interface/src/app/sample/sampleDetail.component.js

```diff
@@ -0,0 +1,70 @@
+angular.module('adage.sampleDetail', [
```

Comment:
Just curious - why is the name of this sub-module `adage.sampleDetail` instead of `adage.sample.detail` or `adage.sample.sampleDetail`, just like `adage.sample.service`? Also, in the long run should `sampleBin` (which is now in analysis) make its way to this `sample` module (something like `adage.sample.sampleBin`)?

Reply:
This is a great question. I went back and forth myself as I thought though this a bit and considered many of the names you did. Here is the thinking that led me to settle on sampleDetail:

* `adage.sample.detail` implies another layer of hierarchy in the source tree (which is not there and I'd like to avoid because those extra layers get tedious fast)

* `adage.sample.service` is a service, not a component, and its naming follows the pattern that the Angular cli uses when generating templates.

Actually, come to think of it, the module should really be called `adage.sampleDetail.component`, which follows the pattern of `adage.sample.service` (and matches the file name to boot).

What do you think? I would change `adage.experimentDetail` to match as well.

Another part of this that doesn't show yet is that some other components we will find useful are `adage.sampleList.component` and `adage.experimentList.component`. I find this naming to be simple and intuitive descriptions of a pattern we will use over and over again.

Reply:
The `sampleBin` is a separate component from what's in `app/sample`. It implements the state management and visual component for the widget in the nav bar that shows how many samples have been selected. This component definitely does not belong in `app/analyze/analysis/` and it is completely overgrown. It needs serious refactoring and will probably become several separate services alongside `adage.sample.service` and `adage.experiment.service`.

Reply:
Ok, do you think we should create an issue for that last piece?

Reply:
I'll make sure there's an issue that covers `sampleBin` refactoring. It's going to happen naturally as we work through #268.